### ISSUE TITLE
Peer panics on restarts

### ISSUE THREAD

**Issue by phahulin** [poanetwork/hydrabadger#2]
After #1 is fixed, when I start all 5 peers they start producing blocks. But when I stop one of them, and then try to start it again, the following [error](https://gist.github.com/phahulin/473b26b9e0678709eb2612623ee9cb62) happens, so it never restarts.
It seems to be a different error, so I open a separate issue

**Comment by c0gent** [comment]
Yes, thanks! Node re-connection is coming later today too (hopefully) :)

Keep the issues coming. Much appreciated.

**Comment by c0gent** [comment]
Besides the fact that reconnecting a node occasionally causes the network to stall, this is fixed. Please let me know if you experience any other problems connecting/reconnecting nodes.
